### PR TITLE
Remove mxnet from default dataset path

### PR DESCRIPTION
**Pull request by abdulfatir**
*Issue #, if available:*

*Description of changes:*
In the spirit of making GluonTS backend agnostic, this PR changes the default dataset download path, removing `.mxnet` from the path.

Old Path: `$HOME/.mxnet/gluon-ts`
New Path: `$HOME/.gluonts`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
Should we get rid of the hyphen as well?

**Comment by abdulfatir** [comment]
Removed hyphen.

File: src/gluonts/dataset/repository/datasets.py [hunk 1]
```diff
@@ -41,15 +40,9 @@ def get_download_path() -> Path:
     -------
     Path
         default path to download datasets or models of gluon-ts.
-        The path is either $MXNET_HOME if the environment variable is defined
-        or /home/username/.mxnet/gluon-ts/
+        The path is $HOME/.gluon-ts/
```
Comment (lostella) [inline src/gluonts/dataset/repository/datasets.py]: Also from here?

Comment (abdulfatir) [reply src/gluonts/dataset/repository/datasets.py]: Fixed, sorry.

**Review by lostella — APPROVED** [review]

**Comment by jaheba** [comment]
We should not just create a folder in the home directory.

There are initiatives to make this more organised: https://wiki.archlinux.org/title/XDG_Base_Directory

We could change this directory to `~/.cache/gluonts`

**Pull request merged**